_posts/2022-06-22-human-centered-code-reviews.md

```diff
@@ -0,0 +1,212 @@
+
+
+## How things were
+In the past releasing new software was a chore. Preparing the release candidate, notifying internal stakeholders and users about it, weeks of manual regression testing and warnings of downtime on the day of release were all part and parcel of the process. Fear of what could go wrong loomed large over the whole process. 
+
+Since the cost of change was so high, *getting things right the first time* was of cardinal importance. A botched deployment meant kicking off the same lengthy process all over again to revert the failed change.
+
+Developers were constrained by the tools and methodologies of the time - being totally confident in the success of a deployment was near impossible. Test driven development, a now widely accepted practice for writing resilient code wasn’t developed till the late 90’s by Kent Beck yet code was already written for devices large and small long before this.
+
+In these times code reviews were a vital tool to ensure code quality. Seasoned engineers would pour over the proposed changes looking for syntax errors, potential compilation issues or any other gaps in code quality that could derail a release and result in a bunch of upset customers or users. It was manual, laborious and the stakes were high if something slipped through.
+
+
+## How things are now
+The tools and methodologies used to ensure quality code have gone from sparse to abundant.
+
+Developers have their choice of rich IDE’s that offer access to a vast plugin ecosystem for most any active programming language, providing convenience features like syntax highlighting and code formatting and more advanced features like code completion. 
+
+Testing frameworks are faster and more feature rich. With a solid testing strategy, developers can have a high degree of confidence that their new code has the intended effect without new bugs or regressions cropping up.
+
+Today’s devs can even provision production-like services on their development machines through techniques like containerisation, providing lean standalone services - the kind of thing that would have been unimaginable for complicated subsystem teams of the past.
+
+For companies that have leaned into continuous delivery and continuous integration as practices, code deployments that once took hours or days could be achieved in mere minutes. Tools like Github Actions can run automated checks against our bundled software before deploying it to production. Some of these checks can include automated tests (unit, e2e, contract etc..), static analysis tools, vulnerability scans just to name a few.
+
+So what does this all mean? 
+
+The need for a second pair of eyes notwithstanding, the code review has gone from being one of a handful of practices to ensure code quality to just one among many. And even at that, it’s prime responsibility of ensuring more than one dev has seen the code is satisfied by the emergency of another rival practice - pair programming.
```

Comment:
```suggestion
The need for a second pair of eyes notwithstanding, the code review has gone from being one of a handful of practices to ensure code quality to just one among many. And even at that, it’s prime responsibility of ensuring more than one dev has seen the code is satisfied by the emergence of another rival practice - pair programming.
```

Is that what you meant? The last sentence is a little tricky to parse for me so maybe reword it?

Reply:
Good spot Will! I'll cover off that typo and see if I can make the sentence a bit more readable